### PR TITLE
FIX-#4022: Fixed empty data frame with index

### LINKED ISSUE #4022: ``dataframe.loc()`` doesn`t work correctly at MODIN_STORAGE_FORMAT=omnisci
When I run:
```
import modin.pandas as pd
import modin.config as cfg
cfg.StorageFormat.put('omnisci')

modin_df = pd.DataFrame(index=range(5))
modin_df.loc[1]
```
I have error:
```
KeyError: 1
```

### PR DESCRIPTION
**Pull request by AndreyPavlenko**
## What do these changes do?
Put index into columns if there are no columns but only index.

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4022 
- [x] tests added and passing
- [ ] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

File: modin/pandas/indexing.py [hunk 1]
```diff
@@ -653,7 +653,7 @@ def __getitem__(self, key):
         --------
         pandas.DataFrame.loc
         """
-        if self.df.empty:
+        if len(self.df.index) == 0 and self.df.empty:
```
Comment (ienkovich) [inline modin/pandas/indexing.py:656]: Why do you need these changes?

Comment (AndreyPavlenko) [reply modin/pandas/indexing.py:656]: If the frame has only index, the following lambda fails with KeyError.

Comment (ienkovich) [reply modin/pandas/indexing.py:656]: Is the problem specific to OmniSci backend or to all backends? Probably, the problem is in backend implementation of `empty`?

Comment (AndreyPavlenko) [reply modin/pandas/indexing.py:656]: Actually, I could find only one implementation of this method - https://github.com/modin-project/modin/blob/master/modin/pandas/dataframe.py#L325

Comment (ienkovich) [reply modin/pandas/indexing.py:656]: I don't understand why making a condition for default to pandas more strict might resolve any issue. We should be able to always be able to default to pandas.

Comment (AndreyPavlenko) [reply modin/pandas/indexing.py:656]: You are right. Actually, the problem is not in default_to_pandas() but in loc[], which is called from lambda. Pandas returns an empty Series for df.loc[1], but modin with omnisci backend fails with KeyError: 'Column F_1 does not exist in schema'. This issue should be fixed instead.

**Review by ienkovich — DISMISSED** [review]
LGTM!

**Review by YarShev — APPROVED** [review]

**Pull request merged**